telegram/utils/helpers.py

```diff
@@ -438,3 +443,79 @@ def __bool__(self):
 
 DEFAULT_NONE = DefaultValue(None)
 """:class:`DefaultValue`: Default `None`"""
+
+
+def get_callback_data_signature(chat_id, callback_data, bot):
+    """
+    Creates a signature, where the key is based on the bots token and username and the message
+    is based on both the chat ID and the callback data.
+
+    Args:
+        chat_id (:obj:`str` | :obj:`int`): The chat the :class:`telegram.InlineKeyboardButton` is
+            sent to.
+        callback_data (:obj:`str`): The callback data.
+        bot (:class:`telegram.Bot`, optional): The bot sending the message.
+
+    Returns:
+        :class:`HMAC`: The encrpyted data to send in the :class:`telegram.InlineKeyboardButton`.
+    """
+    mac = hmac.new('{}{}'.format(bot.token, bot.username).encode('utf-8'),
+                   msg='{}{}'.format(chat_id, callback_data).encode('utf-8'),
+                   digestmod='md5')
+    return mac.digest()
+
+
+def sign_callback_data(chat_id, callback_data, bot):
+    """
+    Prepends a signature based on :meth:`telegram.utils.helpers.get_callback_data_signature`
+    to the callback data.
+
+    Args:
+        chat_id (:obj:`str` | :obj:`int`): The chat the :class:`telegram.InlineKeyboardButton` is
+            sent to.
+        callback_data (:obj:`str`): The callback data.
```

Comment:
`callback_data` here is `id` of the data stored in dict, not the actual `callback_data`. Could we change the name of the argument, I was going through this and got confused why are we returning the actual data

Reply:
Mh, theoretically you could pass string-valued `callback_data`, I guess reformulating to

```
callback_data (:obj:`str`): The callback data/the id of the callback data object.
```
or sim. is appropriate.